app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,31 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @like = Like.new
+  end
+
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new_post, locals: { post: } }
+    end
+  end
+
+  def create
+    post = Post.new(title: params[:post][:title], text: params[:post][:text], user: current_user)
+
+    if post.save
+      flash[:success] = 'Post saved successfully'
+      redirect_to "/users/#{current_user.id}/posts"
+    else
+      flash.now[:error] = 'Error: Post could not be saved'
+      render inline: '<p>Error: Post could not be saved<p>'
+    end
+  end
+
+  def like
+    post = Post.find(params[:id])
+    Like.create(post:, user: current_user)
+    redirect_to "/users/:user_id/posts/#{post[:id]}"
   end
```

Comment:
- [ ] [optional] It would be nice having a params method that should be private to ensure you can reuse it eg for your create and update method which uses the same parameters. Learn more about it [here](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters).